--- a/CMakeLists.txt	2022-06-15 14:29:37.381329575 +0300
+++ b/CMakeLists.txt	2022-06-15 15:06:30.850917318 +0300
@@ -1209,7 +1209,7 @@
 if(RPI4)
     target_link_libraries(${ENGINELIB} "rt")
 endif()
-set_target_properties(${ENGINELIB} PROPERTIES ENABLE_EXPORTS ON LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+set_target_properties(${ENGINELIB} PROPERTIES ENABLE_EXPORTS ON LINK_FLAGS "-Wl,-rpath=/tmp")
 
 add_executable(SeriousSam${MP}
     #${ENGINE_SRCS}
@@ -1303,14 +1303,14 @@
 endif()
 
 if(LINUX)
-    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
     target_link_libraries(SeriousSam${MP} "m")
     target_link_libraries(SeriousSam${MP} "dl")
     target_link_libraries(SeriousSam${MP} "pthread")
     target_link_libraries(SeriousSam${MP} ${SDL2_LIBRARY})
     target_link_libraries(SeriousSam${MP} ${ZLIB_LIBRARIES})
     if(BUILD_DEDICATED_SERVER)
-        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
         target_link_libraries(DedicatedServer${MP} "m")
         target_link_libraries(DedicatedServer${MP} "dl")
         target_link_libraries(DedicatedServer${MP} "pthread")
@@ -1318,7 +1318,7 @@
         target_link_libraries(DedicatedServer${MP} ${ZLIB_LIBRARIES})
     endif()
     if(BUILD_MAKEFONT)
-        set_target_properties(MakeFONT${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+        set_target_properties(MakeFONT${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
         target_link_libraries(MakeFONT${MP} "m")
         target_link_libraries(MakeFONT${MP} "dl")
         target_link_libraries(MakeFONT${MP} "pthread")
@@ -1328,13 +1328,13 @@
 endif()
 
 if(FREEBSD)
-    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
     target_link_libraries(SeriousSam${MP} "m")
     target_link_libraries(SeriousSam${MP} "pthread")
     target_link_libraries(SeriousSam${MP} ${SDL2_LIBRARY})
     target_link_libraries(SeriousSam${MP} ${ZLIB_LIBRARIES})
     if(BUILD_DEDICATED_SERVER)
-        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
         target_link_libraries(DedicatedServer${MP} "m")
         target_link_libraries(DedicatedServer${MP} "pthread")
         target_link_libraries(DedicatedServer${MP} ${SDL2_LIBRARY})
