--- a/CMakeLists.txt	2022-06-15 14:29:37.381329575 +0300
+++ b/CMakeLists.txt	2022-06-15 15:06:30.850917318 +0300
@@ -1239,10 +1239,10 @@
 endif()
 
 if(FREEBSD OR MACOSX)
-set_target_properties(${ENGINELIB} PROPERTIES ENABLE_EXPORTS ON LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+set_target_properties(${ENGINELIB} PROPERTIES ENABLE_EXPORTS ON LINK_FLAGS "-Wl,-rpath=/tmp")
 else()
 target_link_libraries(${ENGINELIB} vulkan)
-set_target_properties(${ENGINELIB} PROPERTIES ENABLE_EXPORTS ON LINK_FLAGS "-Wl,-lvulkan,-rpath,$ORIGIN")
+set_target_properties(${ENGINELIB} PROPERTIES ENABLE_EXPORTS ON LINK_FLAGS "-Wl,-lvulkan,-rpath=/tmp")
 endif()
 
 add_executable(SeriousSam${MP}
@@ -1337,14 +1337,14 @@
 endif()
 
 if(LINUX)
-    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
     target_link_libraries(SeriousSam${MP} "m")
     target_link_libraries(SeriousSam${MP} "dl")
     target_link_libraries(SeriousSam${MP} "pthread")
     target_link_libraries(SeriousSam${MP} ${SDL2_LIBRARY})
     target_link_libraries(SeriousSam${MP} ${ZLIB_LIBRARIES})
     if(BUILD_DEDICATED_SERVER)
-        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
         target_link_libraries(DedicatedServer${MP} "m")
         target_link_libraries(DedicatedServer${MP} "dl")
         target_link_libraries(DedicatedServer${MP} "pthread")
@@ -1352,7 +1352,7 @@
         target_link_libraries(DedicatedServer${MP} ${ZLIB_LIBRARIES})
     endif()
     if(BUILD_MAKEFONT)
-        set_target_properties(MakeFONT${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+        set_target_properties(MakeFONT${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
         target_link_libraries(MakeFONT${MP} "m")
         target_link_libraries(MakeFONT${MP} "dl")
         target_link_libraries(MakeFONT${MP} "pthread")
@@ -1362,13 +1362,13 @@
 endif()
 
 if(FREEBSD)
-    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+    set_target_properties(SeriousSam${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
     target_link_libraries(SeriousSam${MP} "m")
     target_link_libraries(SeriousSam${MP} "pthread")
     target_link_libraries(SeriousSam${MP} ${SDL2_LIBRARY})
     target_link_libraries(SeriousSam${MP} ${ZLIB_LIBRARIES})
     if(BUILD_DEDICATED_SERVER)
-        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath,$ORIGIN")
+        set_target_properties(DedicatedServer${MP} PROPERTIES LINK_FLAGS "-Wl,-rpath=/tmp")
         target_link_libraries(DedicatedServer${MP} "m")
         target_link_libraries(DedicatedServer${MP} "pthread")
         target_link_libraries(DedicatedServer${MP} ${SDL2_LIBRARY})
