diff -aurN b/Engine/Graphics/Vulkan/SvkDynamicBuffers.cpp a/Engine/Graphics/Vulkan/SvkDynamicBuffers.cpp
--- b/Engine/Graphics/Vulkan/SvkDynamicBuffers.cpp	2022-06-15 14:29:37.391329571 +0300
+++ a/Engine/Graphics/Vulkan/SvkDynamicBuffers.cpp	2022-06-15 15:41:53.581613059 +0300
@@ -14,6 +14,8 @@
 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA. */
 
 #include "Engine/StdH.h"
+#include <Engine/Base/Translation.h>
+#include <Engine/Base/Console.h>
 #include <Engine/Graphics/Vulkan/SvkMain.h>
 
 #ifdef SE1_VULKAN
